megatron/mpu/mappings.py

```diff
@@ -98,6 +98,82 @@ def _gather(input_):
     return output
 
 
+def _reduce_scatter_along_seq_dim(input_, seq_dim):
+    """Reduce-scatter the input tensor across model parallel group, scattering across sequence dim."""
+    world_size = get_model_parallel_world_size()
+    # Bypass the function if we are using only 1 GPU.
+    if world_size == 1:
+        return input_
+
+    # Bf16 convert
+    dt = input_.dtype
+    if dt == torch.bfloat16 and get_fp32_allreduce():
```

Comment:
what about fp16?

Reply:
good point!

This actually impacts our other existing comms pre-PR (see e.g. `_reduce`), shall we fix them all in this PR? could add that change in a separate one if preferred.

Reply:
Made the change for all functions! commit (https://github.com/EleutherAI/gpt-neox/pull/1257/commits/ab11a6a71c122cc67440b2206ebe81cfaa70d242) can be cherrypicked into a new PR if preferred.

Reply:
We can keep it in this one